PSKoans/Koans/IntroTraining/AboutDotNet.koans.ps1

```diff
@@ -0,0 +1,152 @@
+using module PSKoans
+[Koan(Position = 103)]
+param()
+<#
+    What is .NET?
```

Comment:
@Sudoblark I've read through this file a few times, and I'm not sure this should be covered here, especially in the introduction.

The intricacies of how PowerShell interacts with the CLR's JIT are... complicated. I'm not convinced we're doing anyone any favours by bringing it in here and giving them the abridged version. If anything, I think it might make it more confusing later.

Knowing about this also doesn't really have a significant impact on how people use PowerShell, generally speaking. It's probably a worthwhile topic much later on in the koans as a bit of background information, but I don't think it is a great topic to hand to beginners. It really needs to be covered in much more detail, but that's probably more fit for a classroom lecture or a blog post than a koan topic, I think?

I think I have to recommend we cut this particular one for now; it might be worth revisiting in a future PR and placing later in the koans if we focus a bit more on the practical side of how PS interacts with .NET -- syntax, particularly useful types to work with, limitations of working with them from PowerShell, etc.

Reply:
The content in this PR is a port of a PowerShell course I use to teach in a facilitator-led format that would usually run as four two-and-a-half hour workshops spread across one to four weeks. In the classroom it worked; any queries or questions and students could get instance feedback and clarification. As a green comment block I wasn't too sure if it would. It's one of the areas I was keen for feedback on in this particular PR.

I would definitely say it's needed at some point when an individual is learning PowerShell, especially if they intended to use it as a stepping stone to learning other technologies such as C# or Python. It would definitely be useful to revisiting at some point, even simple things like being able to use ```[String]::Format()``` are useful. Most of my students started making very basic windows forms, using .NET, as front-ends to their scripts in PowerShell whilst automating all the things.

Would it be worth adding a "Further studying" links as a final Koan, to cover items like this - or other useful learning materials - that we can recommend for further study? If so there's certainly a few items I can port over from my course in a new PR (like the super useful https://underthewire.tech/index.htm) in addition to maybe some material about .NET CLR.

In either case, I can cut it from this PR but I'd be happy to work on something more substantial regarding it in the future?

Reply:
Definitely would say there's value in both options there -- further studying links are applied elsewhere in the koans, and I definitely think there's value in the topic we can explore in another PR. 😄 

It might be worth adding an advice snippet or two mentioning something along these lines to the json file in `PSKoans/Data/Advice` that includes the further reading link when we do that as well. 🙂